src/Graph.elm

```diff
@@ -648,7 +614,7 @@ so that the fold can exit early when the suspended accumulator is not forced.
 fold : (NodeContext n e -> acc -> acc) -> acc -> Graph n e -> acc
 fold f acc graph =
     let
-        go acc graph1 =
+        go acc1 graph1 =
```

Comment:
It has no effect on whether the tail calls can be eliminated, if that is what you mean. I'm not too familiar with Elm's code gen, but I think it would amount to the same code. I mostly wrote it that way to safe characters (probably a bad excuse).

With an optimizing compiler like GHC, there is another perspective, though: The `fold` above is non-recursive and thus can be inlined, which in turn specialises `f` to the concrete argument at the call site. This in turn might enable more optimizations in the call `f ctx acc`, because now `f` is known (e.g. something calling + or whatever). I don't think Elm does anything in that regard.